### PR TITLE
🔨 Enable MSBuild COMM log in bootstrap

### PR DESCRIPTION
**Pull request by jaredpar**
Fixed a gap where we were calling MSBuild without setting up the COMM log. This uses a bigger hammer approach by setting the ENV variables at the YML level.

### CODE REVIEW

File: eng/pipelines/publish-logs.yml [hunk 1]
```diff
@@ -29,4 +29,4 @@ steps:
       targetPath: '$(Build.SourcesDirectory)/artifacts/log/${{ parameters.configuration }}'
       artifactName: '${{ parameters.jobName }} Attempt $(System.JobAttempt) Logs'
     continueOnError: true
-    condition: not(succeeded())
\ No newline at end of file
+    condition: true
```
Comment (jaredpar) [inline eng/pipelines/publish-logs.yml:32]: This is just temporary. I want to see the logs to make sure everything is working the way I expect it to. Will remove before merge.

File: eng/build-utils.ps1 [hunk 2]
```diff
@@ -261,70 +261,71 @@ function Get-PackageDir([string]$name, [string]$version = "") {
...
+# Important to not set $script:bootstrapDir here yet as we're actually in the process of
+# building the bootstrap.
+function Make-BootstrapBuild([switch]$force32 = $false) {
 
-  if ($summary) {
-    $args += " /consoleloggerparameters:Verbosity=minimal;summary"
-  } else {        
-    $args += " /consoleloggerparameters:Verbosity=minimal"
-  }
+  function Run-MSBuild([string]$projectFilePath, [string]$buildArgs = "", [string]$logFileName = "", [string]$configuration = $script:configuration, [switch]$runAnalyzers = $false) {
```
Comment (jaredpar) [inline eng/build-utils.ps1:271]: This is only ever used by the `Make-BootstrapBuild` function so made it a local function to remove confusion around it. Allowed me to delete several parameter's that weren't being used.

**Comment by jaredpar** [comment]
Suggest hiding whitespace for this review

File: eng/pipelines/build-windows-job.yml [hunk 3]
```diff
@@ -34,6 +34,9 @@ jobs:
   timeoutInMinutes: 40
   variables:
     artifactName: ${{ parameters.testArtifactName }}
+    ROSLYNCOMMANDLINELOGFILE: '$(Build.SourcesDirectory)/artifacts/log/${{ parameters.configuration }}/Build.Server.log'
```
Comment (dibarbet) [inline eng/pipelines/build-windows-job.yml:37]: How does the correctness job get these env vars set?  From what I can tell it doesn't use the build template here - https://github.com/dotnet/roslyn/blob/main/azure-pipelines.yml#L275

Comment (jaredpar) [reply eng/pipelines/build-windows-job.yml:37]: Sigh ... it's never easy 😦

File: eng/pipelines/publish-logs.yml (hunk 1 above)
Comment (jaredpar) [inline eng/pipelines/publish-logs.yml:32]: temp change to validate this is working

File: eng/pipelines/build-windows-job.yml (hunk 3 above)
Comment (jaredpar) [reply eng/pipelines/build-windows-job.yml]: @dibarbet okay, refactored this a bit. I added a new template which inserts thevariables and put the template into all the jobs I could find.

**Review by dibarbet — APPROVED** [review]
looks like this is a good candidate for a squash merge 😢 😆

**Comment by jaredpar** [comment]
> looks like this is a good candidate for a squash merge 😢 😆

Yes indeed ... should be the default for all YAML changes :wink:

**Pull request merged**